Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,344 @@
+        /// </summary>
+        public float SurfaceNormalOffset
+        {
+            get => surfaceNormalOffset;
+            set => surfaceNormalOffset = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true, the GameObjectToPlace will remain upright and in line with Vector3.up")]
+        private bool keepOrientationVertical = false;
+
+        /// <summary>
+        /// If true, the GameObjectToPlace will remain upright and in line with Vector3.up
+        /// </summary>
+        public bool KeepOrientationVertical
+        {
+            get => keepOrientationVertical;
+            set => keepOrientationVertical = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.")]
+        private bool debugEnabled = true;
+
+        /// <summary>
+        /// If true and in the Unity Editor, the normal of the raycast hit will be drawn in yellow.
+        /// </summary>
+        public bool DebugEnabled
+        {
+            get => debugEnabled;
+            set => debugEnabled = value;
+        }
+
+        [SerializeField]
+        [Tooltip("If false, the game object to place will not change its rotation according to the surface hit.  The object will" +
+            " remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal" +
+            " if there is a hit.")]
+        private bool rotateAccordingToSurface = false;
+
+        /// <summary>
+        /// If false, the game object to place will not change its rotation according to the surface hit.  The object will 
+        /// remain facing the camera while it is in the placing state.  If true, the object will rotate according to the surface normal
+        /// if there is a hit."
+        /// </summary>
+        public bool RotateAccordingToSurface
+        {
+            get => rotateAccordingToSurface;
+            set => rotateAccordingToSurface = value;
+        }
+
+        [Tooltip("This event is triggered once at the start of the placing state.")]
+        /// <summary>
+        /// This event is triggered once at the start of the placing state.
+        /// </summary>
+        public UnityEvent OnPlacingStarted = new UnityEvent();
+
+        [Tooltip("This event is triggered once when the placing state has ended.")]
+        /// <summary>
+        /// This event is triggered once when the placing state has ended.
+        /// </summary>
+        public UnityEvent OnPlacingStopped = new UnityEvent();
+
+        /// <summary>
+        /// Get the game object layer before the object is placed.  When an object is in the placing state, the layer
+        /// is changed to IgnoreRaycast and then back to the original layer.
+        /// </summary>
+        public int GameObjectLayer { get; protected set; }
```

Comment:
Should this be private (i.e it seems to only be used here). If this is being exposed publicly just for tests, we can also use the internal keyword.